frontend/src/app/auth.service.ts

```diff
@@ -47,6 +47,28 @@ export class AuthService {
         }
       });
   }
+  resetPassword(username: {username: string}): Observable<string> {
+    return this.http.post(
+      this._passwordResetUrl, username)
+      .map(response => {
+        if (response.status === 201) {
+          return 'success';
+        }
+      }).catch(err => Observable.of('error'));
+  }
+
+  confirmResetPassword(
+    password: {password: string},
```

Comment:
If you are not planning to extend the 'password' object with more properties, then use a simple string as a function parameter here.